app/src/main/java/com/ably/tracking/demo/publisher/ably/AssetTrackerImpl.kt

```diff
@@ -17,21 +18,31 @@ import com.ably.tracking.publisher.MapConfiguration
 import com.ably.tracking.publisher.Publisher
 import com.ably.tracking.publisher.RoutingProfile
 import com.ably.tracking.publisher.Trackable
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
 import kotlinx.coroutines.flow.SharedFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 
 class AssetTrackerImpl(
     private val context: Context,
+    coroutineDispatcher: CoroutineDispatcher,
     private val notificationProvider: NotificationProvider,
+    private val locationLogger: LocationLogger,
     private val mapBoxAccessToken: String,
     private val ablyApiKey: String
 ) : AssetTracker {
+
+    private val coroutineScope = CoroutineScope(coroutineDispatcher + Job())
+
     private var publisher: Publisher? = null
 
     override val isConnected: Boolean
         get() = publisher != null
 
-    override fun connect(clientId: String): SharedFlow<Set<Trackable>> {
+    override suspend fun connect(clientId: String): SharedFlow<Set<Trackable>> {
```

Comment:
Why this has changed to a `suspend` function? :thinking:

Reply:
https://github.com/ably/asset-tracking-android-rider-app-demo/pull/34/commits/87c991b122230f891131b877c04bc3622fc99e08